### PR TITLE
Reopen scanner when returning from confirm redemption screen

### PR DESCRIPTION
**Pull request by ivanchukov**

### CODE REVIEW

File: app/src/main/java/org/tokend/template/features/redeem/accept/view/ScanRedemptionActivity.kt [hunk 1]
```diff
@@ -88,6 +88,11 @@ class ScanRedemptionActivity : BaseActivity() {
     override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
         super.onActivityResult(requestCode, resultCode, data)
 
+        if(requestCode == ACCEPT_REQUEST_CODE) {
+            tryOpenQrScanner()
+            return
+        }
+
         if (resultCode == Activity.RESULT_CANCELED) {
             finish()
```
Comment (Radiokot) [inline app/src/main/java/org/tokend/template/features/redeem/accept/view/ScanRedemptionActivity.kt:97]: Reopen scanner no matter what the result was

File: app/src/main/java/org/tokend/template/util/Navigator.kt [hunk 2]
```diff
@@ -452,12 +452,11 @@ class Navigator private constructor() {
         performIntent(intent)
     }
 
-    fun toAcceptRedemption(balanceId: String, request: String) {
+    fun toAcceptRedemption(balanceId: String, request: String, requestCode: Int? = null) {
```
Comment (Radiokot) [inline app/src/main/java/org/tokend/template/util/Navigator.kt]: It it does not finish the source activity then it's name must start with "open"

**Review by Radiokot — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/tokend/template/features/redeem/accept/view/ScanRedemptionActivity.kt (hunk 1 above)
Comment (ivanchukov) [reply app/src/main/java/org/tokend/template/features/redeem/accept/view/ScanRedemptionActivity.kt:97]: I did. This part is to catch scanner closing.

Comment (Radiokot) [reply app/src/main/java/org/tokend/template/features/redeem/accept/view/ScanRedemptionActivity.kt:97]: Sorry

**Pull request merged**